plugins/doc_fragments/_module_csr.py

```diff
@@ -325,6 +326,53 @@ class ModuleDocFragment:
           - privilege_withdrawn
           - aa_compromise
     version_added: 1.4.0
+  custom_extensions:
+    description:
+      - Allows to specify one or multiple custom extensions.
+      - The extension value must not be empty, unless O(custom_extensions[].skip_if_empty) is specified.
+      - In such case the extension will be skipped instead of throwing an error.
+      - Custom extensions require the V(pyasn1) Python package to be installed in the environment where the code is run.
+    type: list
+    elements: dict
+    suboptions:
+      critical:
+        description:
+          - Set the critical flag.
+        type: bool
+        default: false
+      oid:
+        description:
+          - The OID of the custom extension
+          - 'Example: V(1.3.6.1.4.1.34380.1.1.25).'
+        type: str
+      skip_if_empty:
+        description:
+          - Allow empty value to be specified. In such case the extension will be skipped when processing.
+        type: bool
+        default: false
+      value_type:
+        description:
+          - The type of the value. Only valid if O(custom_extensions[].value) is specified.
+          - 'Supported data types are: V(str), V(bool), V(int), V(real).'
+        type: str
+        default: str
+      value:
+        description:
+          - The value of the custom extension.
+          - Mutually exclusive with O(custom_extensions[].value_raw) and O(custom_extensions[].value_b64).
+        type: str
```

Comment:
I'd use
```suggestion
        type: raw
```
here. That way inputs are not forcible converted to strings before they can be handled.